service/history/workflow/mutable_state_impl.go

```diff
@@ -572,6 +573,35 @@ func (e *MutableStateImpl) GetLastWriteVersion() (int64, error) {
 	return common.EmptyVersion, nil
 }
 
+// GetMemo returns the memo of the workflow execution. If the close execution visibility task is incomplete, this method
+// fetches the memo from the mutable state. Otherwise, it fetches the memo from the visibility storage backend.
+func (e *MutableStateImpl) GetMemo(ctx context.Context, visibilityManager manager.VisibilityManager) (*commonpb.Memo, error) {
```

Comment:
You should be able to get VisibilityManager from shard?

Reply:
I wasn't able to find one